# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/websocket/PWebSocketConnection.ipdl
# Commit: 2502449c2e24
# Full Hash: 2502449c2e24993d495c886c717e7f9546cd9e17
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-05 21:44:54
# Regressor Bug: 1497249
# File Overlap Count: 4
# Description:
#   Bug 1497249 - P2: ipdl for nsIWebSocketConnection r=michal
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30624
# ==============================================================================

diff -r 432c7e9dc19f -r 2502449c2e24 netwerk/protocol/websocket/PWebSocketConnection.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/protocol/websocket/PWebSocketConnection.ipdl	Wed Aug 05 15:17:38 2020 +0000
@@ -0,0 +1,33 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set sw=2 ts=8 et tw=80 ft=cpp : */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include protocol PHttpConnectionMgr;
+
+namespace mozilla {
+namespace net {
+
+async refcounted protocol PWebSocketConnection
+{
+  manager PHttpConnectionMgr;
+
+parent:
+  async OnTransportAvailable(nsCString securityInfoSerialization);
+  async OnError(nsresult aStatus);
+  async OnTCPClosed();
+  async OnDataReceived(uint8_t[] aData);
+  async OnUpgradeFailed(nsresult aReason);
+
+child:
+  async EnqueueOutgoingData(uint8_t[] aData);
+  async StartReading();
+  async DrainSocketData();
+
+  async __delete__();
+
+};
+
+} //namespace net
+} //namespace mozilla